_posts/2024-08-30-zarr-v3-specification.markdown

```diff
@@ -0,0 +1,172 @@
+specification from Python-specific concepts, Zarr V3 becomes easier to
+implement in other languages, opening the door for broader adoption in diverse
+computing environments. The specification has also been streamlined, removing
+unnecessary complexities to create a leaner, more focused core that can be
+efficiently implemented across various platforms. 
+
+### Cloud-Native Performance ☁️
+
+Zarr V2 was originally optimized for local file storage, where latency is
+minimal. However, as data storage increasingly moves to the cloud, with its
+higher latency per operation, performance issues have become more apparent. In
+response, Zarr V3 has introduced a restructured approach to metadata storage
+that significantly improves performance in cloud storage environments.
+
+One of the key changes is the consolidation of the `.zarray` and `.zattrs` files
+into a single `zarr.json` file. Previously, `.zarray` contained essential
+information about the array, such as its shape, data type, and chunking, while
+`.zattrs` held custom attributes. Now, this information is combined in
+`zarr.json`, simplifying access and reducing the number of I/O operations
+required.
+
+Additionally, the structure of the array has been optimized. Chunks are now
+grouped into individual directories based on their size, which minimizes the
+overhead of retrieving data from cloud storage. Here's a visual comparison
+between V2 and V3 arrays:
+
+<p align="center">
+ <img src="../assets/images/arrays_v2_v3.png" alt="arrays_v2_v3" width="900">
+ <center> Zarr V2 & V3 Arrays </center>
+</p>
+
+Similarly, the structure of groups has been rethought, with multiple `zarr.json`
+files being used to manage different levels of metadata. The top-level
+`zarr.json` contains basic attributes and node type information, while the
+`zarr.json` files within arrays hold the essential information about the
+arrays. Here's a visual comparison between V2 and V3 groups:
+
+<p align="center">
+ <img src="../assets/images/groups_v2_v3.png" alt="groups_v2_v3" width="900">
+ <center> Zarr V2 & V3 Groups </center>
+</p>
+
+These changes collectively make Zarr V3 far more efficient in high-latency
+environments like cloud storage.
+
+### Increased Extensibility 💪🏻
+
+Zarr's adoption has grown rapidly across various scientific domains, from
+geospatial and bio-imaging to genomics and data science. Each of these fields
+has unique requirements, and Zarr V3 addresses this diversity through a
+extensibility framework.
+
+Extensions in Zarr V3 allow users to add new features and capabilities without
+altering the core specification. This is particularly important for
+accommodating the evolving needs of different communities. For example, the
+extension mechanism lets users manipulate metadata fields, introduce new data
+types, modify the chunk grid to support irregular chunks, and even add new
+codecs.
+
+One exciting new feature made possible by this extensibility is the [sharding
```

Comment:
Except that it's "just a codec", eh?

Reply:
Fixed in https://github.com/zarr-developers/blog/pull/60/commits/7d3e9749df791acc9f51b9f30eab6a5baa612193.